components/exercises/chapter-4-oscar/AList/CityList.test.tsx

```diff
@@ -0,0 +1,102 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import CityList, { ListItem } from './CityList';
+
+describe('ListItem', () => {
+  it('should render the item value', () => {
+    const item = {
+      value: 'Item 1',
+      rest: null,
+    };
+    render(<ListItem item={item} />);
+    const renderedItem = screen.getByText('Item 1');
+    expect(renderedItem).toBeInTheDocument();
+  });
+
+  it('should render "No more cities to visit" if the rest of the list is null', () => {
+    const item = {
+      value: 'Item 1',
+      rest: null,
+    };
+    render(<ListItem item={item} />);
+    const renderedItem = screen.getByText('No more cities to visit');
+    expect(renderedItem).toBeInTheDocument();
+  });
+
+  it('should render the rest of the list if it exists', () => {
+    const item = {
+      value: 'Item 1',
+      rest: {
+        value: 'Item 2',
+        rest: null,
+      },
+    };
+    render(<ListItem item={item} />);
+    const renderedItem = screen.getByText('Item 2');
+    expect(renderedItem).toBeInTheDocument();
+  });
+});
+
+describe('CityList component', () => {
+  test('renders the list of cities', () => {
+    render(<CityList />);
+    expect(screen.getByText('Paris')).toBeInTheDocument();
+  });
+
+  test('displays the countries of the listed cities', () => {
+    render(<CityList />);
+    expect(screen.getByText('France 🇫🇷')).toBeInTheDocument();
+  });
+
+  test('displays the actual dish', () => {
+    render(<CityList />);
+    const mainDishElement = screen.getByTestId('main-dish');
+    expect(mainDishElement).toBeInTheDocument();
```

Comment:
Why don't we grab the element by text content? Grab by test id should be only when you don't have any other choice. Read this https://testing-library.com/docs/queries/about/#priority

Reply:
I also refactored the next 'adds new cities when the "+" button and displays the correct main dish'